### ISSUE TITLE
"azurerm_monitor_diagnostic_setting": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"

### ISSUE THREAD

**Issue by J0F3** [hashicorp/terraform-provider-azurerm#18346]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.22.0

### Affected Resource(s)/Data Source(s)

azurerm_monitor_diagnostic_setting

### Terraform Configuration Files

```hcl
resource "azurerm_monitor_diagnostic_setting" "diag_setting" {
  name                       = "audit-log"
  target_resource_id         = var.resourceid
  log_analytics_workspace_id = var.audit_log_law_id

  log {
    category = "AuditEvent"
    enabled  = true

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  metric {
    category = "AllMetrics"
    enabled  = false
    retention_policy {
      enabled = false
    }
  }
}
```

### Debug Output/Panic Output

```shell
Error: parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
```

### Expected Behaviour

Should work.

Basically same issue as https://github.com/hashicorp/terraform-provider-azurerm/issues/18235.
But it seems it was fixed in 3.22.0 for `azurerm_log_analytics_solution` only. So filing an other issue for `azurerm_monitor_diagnostic_setting` resource which has the same problem which needs a fix as well.

### Actual Behaviour

New configuration (were persumumely no state is present) it  works but for existing configuration / deployment with an existing state terraform apply or plan fails with the error: "parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups""

**Comment by cvs79** [comment]
Any updates on this as this essentially blocks us from moving forward in versions. Last working version is azurerm 3.18

**Comment by J0F3** [comment]
@cvs79 
A workaround I found is to specify the Resouce Id for the Log Analytics Workspace (`var.audit_log_law_id` in the example above) with the casing the provider expects:
e.g.: 
/subscriptions/220f340e-843d-4799-8b2d-99e74b1198d2/resource**G**roups/testing-rg/providers/**M**icrosoft.**O**perational**I**nsights/workspaces/testing-law

instead of 

/subscriptions/220f340e-843d-4799-8b2d-99e74b1198d2/resource**g**roups/testing-rg/providers/**m**icrosoft.**o**perational**i**nsights/workspaces/testing-law

Then delete the Diagnostic Setting in Azure and  run `terraform apply` again with the current version of azurerm. 
The Diagnostic Setting should then get created again with the case sensitive Resouce IDs and the issue should not occur anymore. 

Maybe that helps.
